### PR TITLE
feat(LAB-2609): add LLM_RLHF label export

### PR DESCRIPTION
**Pull request by BlueGrizzliBear**
Corresponding ticket: https://linear.app/kili-technology/issue/LAB-2609/aau-i-export-labels-from-llm-projects

This MR focuses on : 
- Adding `LLM_RLHF` `export_labels` management : needed to convert `ClassicAnnotations` (Classification/Transcription/Ranking split annotations) to return a corresponding `jsonResponse`
- Adding a unit test for the method which converts the split annotations to `jsonResponse` (`_classic_annotations_to_json_response`)
- Adding a unit test for export_labels with `LLM_RLHF` project

### CODE REVIEW

**Review by baptiste-olivier — APPROVED** [review]

File: src/kili/adapters/kili_api_gateway/label/annotation_to_json_response.py [hunk 1]
```diff
@@ -79,10 +89,15 @@ def patch_label_json_response(self, label: Dict, label_id: LabelId) -> None:
...
-            converted_json_resp = _video_label_annotations_to_json_response(
-                annotations=annotations, json_interface=self._project_json_interface
-            )
+            if self._project_input_type == "VIDEO":
+                annotations = cast(List[VideoAnnotation], annotations)
+                converted_json_resp = _video_annotations_to_json_response(
+                    annotations=annotations, json_interface=self._project_json_interface
+                )
+            else:
+                annotations = cast(List[ClassicAnnotation], annotations)
```
Comment (PierreLeveau) [inline src/kili/adapters/kili_api_gateway/label/annotation_to_json_response.py:98]: `ClassicAnnotation` -> `StandardAnnotation`

File: src/kili/adapters/kili_api_gateway/label/annotation_to_json_response.py [hunk 2]
```diff
@@ -147,14 +162,49 @@ def _video_label_annotations_to_json_response(
...
-            raise NotImplementedError(f"Cannot convert annotation to json response: {ann}")
+            raise NotImplementedError(f"Cannot convert video annotation to json response: {ann}")
 
     _add_annotation_metadata(annotations, json_resp)
     _fill_empty_frames(json_resp)
 
     return dict(sorted(json_resp.items(), key=lambda item: int(item[0])))  # sort by frame id
 
 
+def _classic_annotations_to_json_response(
```
Comment (PierreLeveau) [inline src/kili/adapters/kili_api_gateway/label/annotation_to_json_response.py:173]: _standard_annotations_to_json_response

File: src/kili/adapters/kili_api_gateway/label/annotation_to_json_response.py [hunk 3]
```diff
@@ -147,14 +162,49 @@ def _video_label_annotations_to_json_response(
...
+    annotations: List[ClassicAnnotation],
+) -> Dict[str, Dict[JobName, Dict]]:
+    """Convert label annotations to a json response."""
+    json_resp = defaultdict(dict)
+
+    for _, ann in enumerate(annotations):
```
Comment (PierreLeveau) [inline src/kili/adapters/kili_api_gateway/label/annotation_to_json_response.py:179]: `for ann in annotations:`

File: src/kili/adapters/kili_api_gateway/label/annotation_to_json_response.py [hunk 4]
```diff
@@ -147,14 +162,49 @@ def _video_label_annotations_to_json_response(
...
+                )
+
+        elif ann["__typename"] == "TranscriptionAnnotation":
+            ann = cast(TranscriptionAnnotation, ann)
+            ann_json_resp = _transcription_annotation_to_json_response(ann)
+            for job_name, job_resp in ann_json_resp.items():
+                json_resp.setdefault(job_name, {}).setdefault("text", job_resp["text"])
+
+        else:
+            raise NotImplementedError(f"Cannot convert classic annotation to json response: {ann}")
```
Comment (PierreLeveau) [inline src/kili/adapters/kili_api_gateway/label/annotation_to_json_response.py:203]: standard

File: src/kili/adapters/kili_api_gateway/label/annotation_to_json_response.py [hunk 5]
```diff
@@ -226,6 +276,40 @@ def _key_annotations_iterator(annotation: VideoAnnotation) -> Generator:
...
 
 
+def _ranking_annotation_to_json_response(
+    annotation: RankingAnnotation,
+) -> Dict[JobName, Dict]:
+    """Convert ranking annotation to a json response.
+
+    Ranking jobs cannot have child jobs.
+    """
+    json_resp: Dict[JobName, Dict] = defaultdict(dict)
```
Comment (PierreLeveau) [inline src/kili/adapters/kili_api_gateway/label/annotation_to_json_response.py:286]:
```
json_resp = {annotation["job"]: {
     "orders": sorted(...
}}
```
and same below (no need to init to defaultdict

**Review by PierreLeveau — APPROVED** [review]
I have added some name changes suggestions, and some quicker ways to implement.

File: src/kili/adapters/kili_api_gateway/label/annotation_to_json_response.py (hunk 1 above)
Comment (BlueGrizzliBear) [reply src/kili/adapters/kili_api_gateway/label/annotation_to_json_response.py:98]: `ClassicAnnotation` would refer to the type names we have in the front-end & back-end as well, to keep consistency at the differences places of the code.

**Pull request merged**